### ISSUE TITLE
I need info about your library

### ISSUE THREAD

**Issue by orestes22** [elsa-workflows/elsa-designer#16]
Hello, Im trying to add you library to my UI project. Im using NextJs, this is SSR of ReactJs. Also Im using material-ui framework.

Im following your instructions and I cannt make your library works with my application. 

I made this example https://codesandbox.io/s/elsa-designer-ficqu 

I need more information or documentation of how integrate your library to a Reactjs application and where is the library API info to have the possibility to manage your library. 

Best Regards
Orestes

**Comment by sfmskywalker** [comment]
Hi @orestes22 ,

I did a little bit of research, and it turns out the [recommended approach](https://stenciljs.com/docs/react) from the Stencil team is for me to provide ReactJS wrappers around the components. I started digging into that, but ran into an issue where the generated wrapper wouldn't compile because of a dependency on anpther Stencil component library I'm using ([bs-components](https://bs-components.github.io/)).

Long story short, I need a little bit of time to solve this and provide proper ReactJS wrappers, which will live here: https://github.com/elsa-workflows/elsa-designer-react.

Same goes for the library API documentation - it will come, but it will take a bit of time.

Thanks,
Sipke

**Comment by orestes22** [comment]
Hi, @sfmskywalker , I appreciate a lot your time into this. 

I having making a research looking libraries in React that can create workflows and i found 3 o 4(there isnt to much) but your project is the only one that have a complete solution, you have a backend and frontend. 
But I kindly recommend for your free time, try to create version of frontend in React or/and Angular, without Stencil. Also without a design framework. If the people want a design they can pass it to the component like props. 
If you accomplish that you will a complete solutions to manage workflows. Nobody have that at least in ReactJs. 

Probably you can charge for support like jsplumb do in there website and have some profits from your work. 

But follow in Stencil  to accomplish something can limited your library future when there is outside NextJs , Preact , and others. Those are compatible with Reactjs but Im not sure with Stencil.   

Best Regards
Orestes

**Comment by sfmskywalker** [comment]
Thanks for the feedback @orestes22 , I really appreciate it.
Regarding the creation of a design-free component, that is on my list for future improvements, because I do want users to be able to customize that.

Regarding StencilJS, it's a web component compiler, which means it outputs HTML5 web components, as if they were native components like `<div>` and `<textarea>` etc. The reason I went with that instead of a library like React or Angular is so that I do not force anyone onto that library - on the contrary, users are now able to choose whatever library or framework they like best, be it React, Angular, Vue, Svelte, or even vanilla JS,

So please rest assured, the goal of this component is so that anyone can use it with their framework of choice. I just need time to make it so.
